File: node/src/pos-integration/services/ty-bank/ty-bank-pos.service.ts
import { Injectable } from "@nestjs/common";
import { PosService } from "../pos.service";
import { RefundRequest } from "../../model/refund/refund-request";
import { RefundResponse } from "../../model/refund/refund-response";
import { AuthRequest } from "../../model/auth/auth-request";
import { RedirectUrlRequest } from "../../model/redirect-url/redirect-url-request";
import { RedirectUrlResponse } from "../../model/redirect-url/redirect-url-response";
import { AuthResponse } from "../../model/auth/auth-response";

@Injectable()
export class TyBankPosService implements PosService {

  public async refund(request: RefundRequest): Promise<RefundResponse> {
    // TODO: business implementation section
    // "refund" flow provides to refund the amount to user via TyBank virtual pos.
    // The user will see refunded amount in his or her account in 1-2 days.
    return null;
  }

  public async auth(request: AuthRequest): Promise<AuthResponse> {
    // TODO: business implementation section
    // "auth" flow enables the user to make payment via credit card from TyBank virtual pos.
    // This payment method captures the amount from the user's credit card directly.
    return null;
  }

  public async fetchRedirectionUrl(request: RedirectUrlRequest): Promise<RedirectUrlResponse> {
    return null;
  }
}

File: kotlin/src/main/kotlin/com/trendyol/posintegrationapi/service/tybank/TyBankPosService.kt
package com.trendyol.posintegrationapi.service.tybank

import com.trendyol.posintegrationapi.exception.PosException
import com.trendyol.posintegrationapi.model.auth.AuthRequest
import com.trendyol.posintegrationapi.model.auth.AuthResponse
import com.trendyol.posintegrationapi.model.redirect_url.RedirectUrlRequest
import com.trendyol.posintegrationapi.model.redirect_url.RedirectUrlResponse
import com.trendyol.posintegrationapi.model.refund.RefundRequest
import com.trendyol.posintegrationapi.model.refund.RefundResponse
import com.trendyol.posintegrationapi.service.PosService

class TyBankPosService : PosService {
    override fun refund(request: RefundRequest): RefundResponse? {
        // TODO: business implementation section
        // "refund" flow provides to refund the amount to user via TyBank virtual pos.
        // The user will see refunded amount in his or her account in 1-2 days.
        return null
    }

    override fun auth(request: AuthRequest): AuthResponse? {
        // TODO: business implementation section
        // "auth" flow enables the user to make payment via credit card from TyBank virtual pos.
        // This payment method captures the amount from the user's credit card directly.
        return null
    }

    override fun fetchRedirectionUrl(request: RedirectUrlRequest): RedirectUrlResponse? {
        return null
    }
}
